ddl/ddl.go

```diff
@@ -371,9 +371,13 @@ func (d *ddl) start(ctx context.Context, ctxPool *pools.ResourcePool) {
 		for _, worker := range d.workers {
 			worker.wg.Add(1)
 			w := worker
-			go util2.WithRecovery(ctx, func(ctx context.Context) {
-				w.start(d.ddlCtx)
-			}, util2.WithPanicLogf("[ddl-%s] ddl %s", w, d.uuid), util2.WithPanicMetricInc(metrics.LabelDDL))
+			go util2.WithRecovery(ctx, func(ctx context.Context) { w.start(d.ddlCtx) },
+				func(r interface{}) {
+					if r != nil {
```

Comment:
Can we remove the check, only call this function when `r` is not nil?

Reply:
maybe we can't, just like this case https://github.com/pingcap/tidb/pull/7666#discussion_r216570580

some operation(e.g. `close(xxChan)`) need do `r == nil`  too